src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -2010,6 +2010,12 @@ class AlterTableStatementSegment(BaseSegment):
                         optional=True,
                     ),
                 ),
+                # Rename
```

Comment:
I think some existing dialects already have similar support. Should we look at removing/consolidating those here?

I didn't look closely, but I did a code search and saw this:
```
src/sqlfluff/dialects/dialect_teradata.py
187:    RENAME TABLE OLD_TABLENAME (TO|AS) NEW_TABLENAME
192:        "RENAME",

src/sqlfluff/dialects/dialect_exasol.py
1033:    """ALTER TABLE RENAME.."""
1037:        "RENAME",
1112:                "RENAME",
1182:# RENAME
1186:    """`RENAME` statement.
1197:    match_grammar = StartsWith("RENAME")
1199:        "RENAME",

src/sqlfluff/dialects/dialect_snowflake.py
614:            Sequence("RENAME", "TO", Ref("ObjectReferenceSegment")),
```

Reply:
It looks like exasol doesn't support the `alert table x rename to y` syntax: https://docs.exasol.com/sql/alter_table(column).htm?tocpath=SQL%20Reference%7CSQL%20Statements%7CDefinition%20of%20Database%20(DDL)%7C_____7

I'm going to keep this separate for now. Potentially to be revisited though if the need arises 😁